File: Expenses/Expenses.ApiRepository/Repository.cs
using Expenses.ApiRepository.Interfaces;
using Expenses.ApiRepository.Repositories;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Expenses.ApiRepository
{
    public class Repository
    {
        private const string Path = "api";

        public Repository(string host)
        {
            UsersRepository = new Users(host, Path);
            HouseholdsRepository = new Households(host, Path);
            MessagesRepository = new Messages(host, Path);
            CategoriesRepository = new Categories(host, Path);
            CashFlowsRepository = new CashFlows(host, Path);
        }

        public IUsers UsersRepository { get; }
        public IHouseholds HouseholdsRepository { get; }
        public IMessages MessagesRepository { get; }
        public ICategories CategoriesRepository { get; }
        public ICashFlows CashFlowsRepository { get; }
    }
}

File: Expenses.AndroidClient/Expenses.Repository/src/main/java/net/azurewebsites/expenses_by_klaudia/repository/Repository.java
package net.azurewebsites.expenses_by_klaudia.repository;

public class Repository {
    private final String Path = "api";
    public Repository(String host) {
        _usersRepository = new Users(host, Path);
        _categoriesRepository = new Categories(host, Path);
        _cashFlowsRepository = new CashFlows(host, Path);
        _messagesRepository = new Messages(host, Path);
        _householdsRepository = new Households(host, Path);
    }

    private Users _usersRepository;
    public Users GetUsersRepository() {
        return _usersRepository;
    }

    private Categories _categoriesRepository;
    public Categories GetCategoriesRepository() {
        return _categoriesRepository;
    }

    private CashFlows _cashFlowsRepository;
    public CashFlows getCashFlowsRepository() {
        return _cashFlowsRepository;
    }

    private Messages _messagesRepository;
    public Messages getMessagesRepository() {
        return _messagesRepository;
    }

    private Households _householdsRepository;
    public Households getHouseholdsRepository() {
        return _householdsRepository;
    }
}
